internal/publish/publish.go

```diff
@@ -0,0 +1,180 @@
+	if err != nil {
+		return errors.Wrap(err, "can't copy over the updated package")
+	}
+
+	fmt.Println("Push new package revision to storage")
+	err = storage.PushChanges(githubUser, r, destination)
+	if err != nil {
+		return errors.Wrapf(err, "pushing changes failed")
+	}
+
+	if skipPullRequest {
+		fmt.Println("Skip opening a new pull request")
+		return nil
+	}
+
+	fmt.Println("Check if pull request is already open")
+	alreadyOpen, err := checkIfPullRequestAlreadyOpen(githubClient, *m)
+	if err != nil {
+		return errors.Wrapf(err, "can't check if pull request is already open")
+	}
+	if alreadyOpen {
+		fmt.Println("Pull request with package update is already open")
+		return nil
+	}
+
+	fmt.Println("Open new pull request")
+	err = openPullRequest(githubClient, githubUser, destination, *m, commitHash)
+	if err != nil {
+		return errors.Wrapf(err, "can't open a new pull request")
+	}
+	return nil
+}
+
+func findLatestPackageRevision(r *git.Repository, packageName string) (*storage.PackageVersion, string, error) {
+	var revisions storage.PackageVersions
+
+	revisionStageMap := map[string]string{}
+	for _, currentStage := range []string{productionStage, stagingStage, snapshotStage} {
+		logger.Debugf("Change stage to %s", currentStage)
+
+		err := storage.ChangeStage(r, currentStage)
+		if err != nil {
+			return nil, "", errors.Wrapf(err, "can't change stage to %s", currentStage)
+		}
+
+		logger.Debugf("Find revisions of the \"%s\" package", packageName)
+		revs, err := storage.ListPackagesByName(r, packageName)
+		if err != nil {
+			return nil, "", errors.Wrapf(err, "can't list packages")
+		}
+
+		for _, rev := range revs {
+			logger.Debugf("Found package revision: %s", rev.String())
+			revisionStageMap[rev.String()] = currentStage
+		}
+		revisions = append(revisions, revs...)
+	}
+
+	if len(revisions) == 0 {
+		logger.Debugf("No published revisions of the \"%s\" package so far. It seems to be brand new.", packageName)
+		return nil, "", nil
+	}
+
+	revisions = revisions.FilterPackages(true)
+	latest := revisions[0]
+	return &latest, revisionStageMap[latest.String()], nil
+}
+
+func copyLatestRevisionIfAvailable(r *git.Repository, latestRevision *storage.PackageVersion, stage string, manifest *packages.PackageManifest) (string, error) {
+	nonce := time.Now().UnixNano()
+	destinationBranch := fmt.Sprintf("update-%s-%d", snapshotStage, nonce)
+	err := storage.CopyPackagesWithTransform(r, stage, snapshotStage, optionalPackageVersions(latestRevision), destinationBranch, createRewriteResourcePath(manifest))
+	if err != nil {
+		return "", errors.Wrap(err, "can't copy latest revision")
+	}
+	return destinationBranch, nil
+}
+
+func optionalPackageVersions(pv *storage.PackageVersion) storage.PackageVersions {
+	if pv != nil {
+		return storage.PackageVersions{*pv}
+	}
+	return storage.PackageVersions{}
+}
+
+func createRewriteResourcePath(manifest *packages.PackageManifest) func(string, []byte) (string, []byte) {
```

Comment:
WDYT about making this a method of `packages.PackageManifest`?

Reply:
Ignore this suggestion. I think this is fine where it is, given its limited usage scope.